src/vario/ui/settings/settings.cpp

```diff
@@ -289,30 +292,81 @@ void Settings::adjustContrast(Button dir) {
 void Settings::adjustSinkAlarm(Button dir) {
   uint16_t* sound = fx_neutral;
 
+  // first find index of best-matching setting in the valid options array
+  uint8_t index = 0;
+  for (uint8_t i = 0; i < sizeof(sinkAlarmOptions_[vario_sinkAlarm_units ? 1 : 0]); i++) {
+    if (vario_sinkAlarm >= -0.1f + sinkAlarmOptions_[vario_sinkAlarm_units ? 1 : 0][i]) {
+      index = i;
+      break;
+    }
+  }
```

Comment:
nit: I'd probably evaluate like this:

```suggestion
  uint8_t index = 0;
  float min_err = 1e9f;
  for (uint8_t i = 0; i < sizeof(sinkAlarmOptions_[vario_sinkAlarm_units ? 1 : 0]); i++) {
    float err = abs(vario_sinkAlarm - sinkAlarmOptions_[vario_sinkAlarm_units ? 1 : 0][i]);
    if (err < min_err) {
      index = i;
      min_err = err;
    }
  }
```

Reply:
thanks, done!